src/cascadia/PublicTerminalCore/Source.cpp

```diff
@@ -0,0 +1,291 @@
+
+void _stdcall SendTerminalOutput(void* terminal, LPCWSTR data)
+{
+    auto publicTerminal = (HwndTerminal*)terminal;
+    publicTerminal->SendOutput(data);
+}
+
+HRESULT _stdcall TriggerResize(void* terminal, double width, double height, _Out_ int* charColumns, _Out_ int* charRows)
+{
+    auto publicTerminal = (HwndTerminal*)terminal;
+    return publicTerminal->Refresh(width, height, charColumns, charRows);
+}
+
+void _stdcall DpiChanged(void* terminal, int newDpi)
+{
+    auto publicTerminal = (HwndTerminal*)terminal;
+    publicTerminal->_UpdateFont(newDpi);
+}
+
+void _stdcall UserScroll(void* terminal, int viewTop)
+{
+    auto publicTerminal = (HwndTerminal*)terminal;
+    publicTerminal->_terminal->UserScrollViewport(viewTop);
+}
+
+void _stdcall StartSelection(void* terminal, COORD cursorPosition, bool altPressed)
+{
+    COORD terminalPosition = {
+        static_cast<SHORT>(cursorPosition.X),
+        static_cast<SHORT>(cursorPosition.Y)
+    };
+
+    auto publicTerminal = (HwndTerminal*)terminal;
+    const auto fontSize = publicTerminal->_actualFont.GetSize();
+
+    terminalPosition.X /= fontSize.X;
+    terminalPosition.Y /= fontSize.Y;
+
+    publicTerminal->_terminal->SetSelectionAnchor(terminalPosition);
+    publicTerminal->_terminal->SetBoxSelection(altPressed);
+
+    publicTerminal->_renderer->TriggerSelection();
+}
+
+void _stdcall MoveSelection(void* terminal, COORD cursorPosition)
+{
+    COORD terminalPosition = {
+        static_cast<SHORT>(cursorPosition.X),
+        static_cast<SHORT>(cursorPosition.Y)
+    };
+
+    auto publicTerminal = (HwndTerminal*)terminal;
+    const auto fontSize = publicTerminal->_actualFont.GetSize();
+
+    terminalPosition.X /= fontSize.X;
+    terminalPosition.Y /= fontSize.Y;
+
+    publicTerminal->_terminal->SetEndSelectionPosition(terminalPosition);
+    publicTerminal->_renderer->TriggerSelection();
+}
+
+void _stdcall ClearSelection(void* terminal)
+{
+    auto publicTerminal = (HwndTerminal*)terminal;
+    publicTerminal->_terminal->ClearSelection();
+}
+bool _stdcall IsSelectionActive(void* terminal)
+{
+    auto publicTerminal = (HwndTerminal*)terminal;
+    bool selectionActive = publicTerminal->_terminal->IsSelectionActive();
+    return selectionActive;
+}
+
+// Copies the selected text into the clipboard.
+const wchar_t* _stdcall GetSelection(void* terminal)
+{
+    auto publicTerminal = (HwndTerminal*)terminal;
+
+    std::wstring selectedText = publicTerminal->_terminal->RetrieveSelectedTextFromBuffer(false);
+
+    const wchar_t* text = selectedText.c_str();
+    size_t textChars = wcslen(text) + 1;
+    size_t textBytes = textChars * sizeof(wchar_t);
+    wchar_t* returnText = NULL;
+
+    returnText = (wchar_t*)::CoTaskMemAlloc(textBytes);
+
+    if (returnText == nullptr)
+    {
+        return nullptr;
+    }
+
+    wcscpy_s(returnText, textChars, text);
+
+    ClearSelection(terminal);
+
+    return returnText;
+}
+
+void _stdcall DestroyTerminal(void* terminal)

```

Comment:
This is scary. You're going to just straight up delete any pointer you're given?

Reply:
What would you recommend for cleaning up the `HwndTerminal`?